src/remote/watcher.coffee

```diff
@@ -77,8 +79,7 @@ class RemoteWatcher
                         @whenReady callback
             else
                 @changes = @couch.client.changes
-                    filter: (doc) ->
-                        doc.docType?.toLowerCase() in ['file', 'folder']
+                    filter: @filterName
```

Comment:
My bad. I thought filters worked only in the local database and only views worked remotely, but I was wrong. After reading pouchdb doc again, I see that `filter` can also work with a remote design document. It's only with a filter that is a raw function (not in a design document) that the replication transfers all the records and filter them client-side.